src/librustc_save_analysis/lib.rs

```diff
@@ -144,7 +145,20 @@ impl<'l, 'tcx> SaveContext<'l, 'tcx> {
                     span: self.span_from_span(item.ident.span),
                     name: item.ident.to_string(),
                     qualname,
-                    value: make_signature(decl, generics),
+                    value: fn_to_string(
+                        decl,
+                        hir::FnHeader {
```

Comment:
That's unfortunate, together with importing `rustc_target` as a dependency...

I think this would be alleviated if we stored `FnSig` that has both `FnHeader` and `FnDecl` in `ForeignItemKind::Fn` and accessed that directly rather than just synthesizing the header (we lose the extern ABI information, in particular). The header is small and `Copy` so hopefully this doesn't impact things much...

Would such a change to HIR make sense? @RalfJung since you've already invested a bit of time in this PR series 😅

Reply:
Sorry I know nothing about HIR.^^

Reply:
Let's try then... @petrochenkov